### PR TITLE
Add examples of index and constraint involving multiple properties.

### PR DESCRIPTION
**Pull request by vpetrovykh**
Add examples of indexes and constraints that use arbitrary expressions
and multiple properties.

Issue #1609

### CODE REVIEW

File: docs/datamodel/constraints.rst [hunk 1]
```diff
@@ -39,14 +39,28 @@ The standard library defines the following constraints:
...
+    Example of using an ``expression`` constraint based on a couple of
+    object properties to restrict maximum magnitude for a vector:
+
+    .. code-block:: sdl
+
+        type Vector {
+            required property x -> float64;
+            required property y -> float64;
+            constraint expression on (
+                __subject__.x^2 + __subject__.y^2 < 25
```
Comment (elprans) [inline docs/datamodel/constraints.rst:60]:
```suggestion
                (.x^2 + .y^2) < 25
```

File: docs/edgeql/ddl/constraints.rst [hunk 2]
```diff
@@ -339,6 +334,18 @@ constraint:
...
 
+Create a Vector with a maximum magnitude:
+
+.. code-block:: edgeql
+
+    CREATE TYPE Vector {
+        CREATE REQUIRED PROPERTY x -> float64;
+        CREATE REQUIRED PROPERTY y -> float64;
+        CREATE CONSTRAINT expression ON (
+            __subject__.x^2 + __subject__.y^2 < 25
```
Comment (elprans) [inline docs/edgeql/ddl/constraints.rst:345]:
```suggestion
            (.x^2 + .y^2) < 25
```

File: docs/edgeql/sdl/constraints.rst [hunk 3]
```diff
@@ -30,6 +30,18 @@ Declare a *concrete* constraint on an integer type:
...
 
+Declare a *concrete* constraint on an object type:
+
+.. code-block:: sdl
+
+    type Vector {
+        required property x -> float64;
+        required property y -> float64;
+        constraint expression on (
+            __subject__.x^2 + __subject__.y^2 < 25
```
Comment (elprans) [inline docs/edgeql/sdl/constraints.rst:41]:
```suggestion
            (.x^2 + .y^2) < 25
```

File: docs/datamodel/constraints.rst (hunk 1 above)
Comment (vpetrovykh) [reply docs/datamodel/constraints.rst:60]: There's actually a bug preventing that right now.

**Pull request merged**